### PR TITLE
[jit] Add LSTMCell backward pass expect tests

### PR DESCRIPTION
**Pull request by zou3519**
- Exposed get_debug_graph for ScriptModule (gets the debug graph for its
  forward Method)
- Added forward/backward expect tests for lstm and milstm cells. These
  are intended to prevent regressions

cc @apaszke @zdevito

### CODE REVIEW

File: torch/csrc/jit/init.cpp [hunk 1]
```diff
@@ -120,6 +120,9 @@ void initJITBindings(PyObject *module) {
       });
 
   py::class_<ExecutionPlanState>(m, "ExecutionPlanState")
+    .def("_graph_copy", [](ExecutionPlanState& s) {
+      return s.graph->copy();
+    })
```
Comment (apaszke) [inline torch/csrc/jit/init.cpp]: Can we make `assertExpectedGraph` *not* mutate the inputs? That's very counterintuitive, and shouldn't really happen. Let's just make it clone the graph when it's called (and expose `Graph::copy()` if it's not exposed yet).

File: test/test_jit.py [hunk 2]
```diff
@@ -107,6 +157,27 @@ def get_fn(file_name, script_path):
     return fn
 
 
+def get_execution_plan(graph_executor):
```
Comment (apaszke) [inline test/test_jit.py]: This is really the debug dumb and not the graph executor itself, right?

File: test/expect/TestScript.test_lstm_fusion_cuda-backward.expect [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  %30 : Float(3, 20) = aten::neg(%ingate)
+  %31 : Float(3, 20) = aten::add(%30, %19, %19)
+  %32 : Float(3, 80) = prim::FusionGroup_3[device=0](%31, %ingate, %29, %forgetgate, %27, %25, %outgate, %22, %cx, %23, %cellgate, %0, %18)
+  %33 : Float(3, 80) = aten::mul(%32, %19)
+  %34 : Float(80!, 3!) = aten::t(%33)
+  %35 : Float(80, 20) = aten::mm(%34, %hx)
+  %36 : Float(80!, 3!) = aten::t(%32)
+  %37 : Float(80, 10) = aten::mm(%36, %x)
+  return (%37, %35, %33, %33);
+}
```
Comment (apaszke) [inline test/expect/TestScript.test_lstm_fusion_cuda-backward.expect]: Uh so many missed fusion opportunities 😄Looking forward to your patches landing.

**Review by apaszke — APPROVED** [review]
We should avoid mutation in `assert*` functions. Once this is fixed, LGTM.

File: test/test_jit.py (hunk 2 above)
Comment (zou3519) [reply test/test_jit.py]: Yes, I'll change the name to reflect that

**Comment by zou3519** [comment]
@apaszke I was mistaken, test_jit's assert* functions do not mutate the graph. The jit passes that the assert* functions run however require std::shared_ptr<Graph> as an input instead of the Graph* given by the debug dump, so I'll add a copy to the graph obtained from the debug dump.

**Pull request closed without merging**